web/src/components/TaxRegistrationModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { OnboardingOwnership } from "./onboarding/OnboardingOwnership";
+import { OnboardingTaxId } from "./onboarding/OnboardingTaxId";
+
+interface Props {
+  isOpen: boolean;
+  close: () => void;
+  onSave: (
+    newValue: TaskProgress,
+    userData: UserData,
+    { redirectOnSuccess }: { redirectOnSuccess: boolean }
+  ) => void;
+}
+
+export const TaxRegistrationModal = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+  const { userData, update } = useUserData();
+  const [profileData, setProfileData] = useState<ProfileData>(createEmptyProfileData());
+  const [fieldStates, setFieldStates] = useState<ProfileFieldErrorMap>(createProfileFieldErrorMap());
+
+  useEffect(() => {
+    if (!userData) return;
+    setProfileData(userData.profileData);
+  }, [userData]);
+
+  const onValidation = (field: ProfileFields, invalid: boolean) => {
+    setFieldStates({ ...fieldStates, [field]: { invalid } });
+  };
+
+  const onSubmit = async () => {
+    if (!userData) return;
+    const errorMap = {
+      ...fieldStates,
+      businessName: { invalid: !profileData.businessName },
+      existingEmployees: { invalid: !profileData.existingEmployees },
+      OnboardingTaxId: { invalid: !profileData.taxId },
+    };
+    setFieldStates(errorMap);
+    if (Object.keys(errorMap).some((k) => errorMap[k as ProfileFields].invalid)) return;
+
+    const updatedUserData = { ...userData, profileData };
+    await update(updatedUserData);
+    props.onSave("COMPLETED", updatedUserData, { redirectOnSuccess: true });
+    props.close();
+  };
+
+  const showBusinessField = () =>
+    userData?.profileData.businessName === "" &&
+    LookupLegalStructureById(userData?.profileData.legalStructureId).requiresPublicFiling;
+
+  const showTaxIdField = () =>
+    LookupLegalStructureById(userData?.profileData.legalStructureId).requiresPublicFiling;
+
+  return (
+    <ProfileDataContext.Provider
+      value={{
+        state: {
+          profileData: profileData,
+          flow: "STARTING",
+          municipalities: [],
+        },
+        setProfileData,
+        setUser: () => {},
+        onBack: () => {},
+      }}
+    >
+      <ModalTwoButton
+        isOpen={props.isOpen}
+        close={props.close}
+        title={Config.taxRegistrationModal.title}
+        maxWidth={"md"}
+        primaryButtonText={Config.taxRegistrationModal.saveButtonText}
+        primaryButtonOnClick={onSubmit}
+        secondaryButtonText={Config.taxRegistrationModal.cancelButtonText}
+      >
+        <div className="margin-bottom-3">
+          <Content>{Config.taxRegistrationModal.subtitle}</Content>
+        </div>
+        <hr />
+        {showBusinessField() ? (
+          <OnboardingBusinessName onValidation={onValidation} fieldStates={fieldStates} />
+        ) : (
+          <></>
+        )}
+        {showTaxIdField() ? <OnboardingTaxId onValidation={onValidation} fieldStates={fieldStates} /> : <></>}
```

Comment:
these could both be just `{showTaxIdField() && <OnboardingTaxId onValidation={onValidation} fieldStates={fieldStates} />}`, we don't need a ternary with an empty case